src/elvis_webhook.erl

```diff
@@ -1,56 +1,25 @@
 -module(elvis_webhook).
 
--export([event/2]).
-
--export_type([request/0]).
+-behaviour(egithub_webhook).
 
--type event() :: pull_request.
--type request() :: #{headers => map(), body => map()}.
-
--type github_info() ::
-        {
-          egithub:credentials(),
-          egithub:repository(),
-          integer(),
-          [map()]
-        }.
+-export([event/2]).
+-export([handle_pull_request/3]).
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-%%% Public API
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-
--spec event(egithub:credentials(), request()) -> ok | {error, term()}.
-event(Cred, #{headers := Headers, body := Body}) ->
-    HeaderName = <<"x-github-event">>,
-    case maps:is_key(HeaderName, Headers) of
-        false ->
-            {error, missing_header};
-        true ->
-            EventName = maps:get(HeaderName, Headers),
-            EventData = jiffy:decode(Body, [return_maps]),
-            Config = elvis_config:default(),
-            event(Config, Cred, EventName, EventData)
-    end.
+%%% External Functions
 
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-%%% Private functions
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+-spec event(egithub:credentials(), egithub_webhook:request()) ->
+  ok | {error, term()}.
+event(Cred, Request) -> egithub_webhook:event(?MODULE, Cred, Request).
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-%%% Events
+%%% Callbacks
 
--spec event(elvis_config:config(),
-            egithub:credentials(),
-            event(),
-            map()) -> ok | {error, term()}.
-event(LocalConfig,
-      Cred,
-      <<"pull_request">>,
-      #{<<"number">> := PR, <<"repository">> := Repository}) ->
-    Repo = binary_to_list(maps:get(<<"full_name">>, Repository)),
-    Config = repo_config(Cred, Repo, LocalConfig),
-
-    {ok, GithubFiles} = egithub:pull_req_files(Cred, Repo, PR),
+-spec handle_pull_request(
+  egithub:credentials(), string(), [egithub_webhook:file()]) ->
+  {ok, [egithub_webhook:messge()]} | {error, term()}.
```

Comment:
Typo: should be `message` instead of `messge`. The typo os also there on erlang-github [here](https://github.com/inaka/erlang-github/blob/master/src/egithub_webhook.erl#L14), I missed it in the PR :disappointed: